stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -0,0 +1,365 @@
+    }
+
+    public long getEndMs() {
+      return mEndMs;
+    }
+
+    public synchronized void mergeThreadResult(FuseIOTaskResult.ThreadCountResult threadResult) {
+      if (mThreadCountResult == null) {
+        mThreadCountResult = threadResult;
+      } else {
+        try {
+          mThreadCountResult.merge(threadResult);
+        } catch (Exception e) {
+          mThreadCountResult.addErrorMessage(e.getMessage());
+        }
+      }
+    }
+
+    public synchronized FuseIOTaskResult.ThreadCountResult getResult() {
+      return mThreadCountResult;
+    }
+  }
+
+  private final class BenchThread implements Callable<Void> {
+    private final BenchContext mContext;
+    private final List<String> mFilesPath;
+    private final int mThreadId;
+    private final byte[] mBuffer;
+    private final long mFileSize;
+
+    private FileInputStream mInStream = null;
+    private FileOutputStream mOutStream = null;
+
+    private final FuseIOTaskResult.ThreadCountResult mThreadCountResult =
+        new FuseIOTaskResult.ThreadCountResult();
+
+    private BenchThread(BenchContext context, int threadId, int numThreads) {
+      mContext = context;
+      mThreadId = threadId;
+      mFilesPath = new ArrayList<>();
+      for (int i = mThreadId; i < mParameters.mNumFiles; i += numThreads) {
+        mFilesPath.add(mParameters.mLocalPath + "/data-" + i);
+      }
+
+      mBuffer = new byte[(int) FormatUtils.parseSpaceSize(mParameters.mBufferSize)];
+      Arrays.fill(mBuffer, (byte) 'A');
+
+      mFileSize = FormatUtils.parseSpaceSize(mParameters.mFileSize);
+    }
+
+    @Override
+    public Void call() {
+      try {
+        runInternal();
+      } catch (Exception e) {
+        LOG.error(Thread.currentThread().getName() + ": failed", e);
+        mThreadCountResult.addErrorMessage(e.getMessage());
+      }
+
+      // Update thread count result
+      mThreadCountResult.setEndMs(CommonUtils.getCurrentMs());
+      mContext.mergeThreadResult(mThreadCountResult);
+
+      return null;
+    }
+
+    private void runInternal() throws Exception {
+      // When to start recording measurements
+      long recordMs = mContext.getStartMs() + FormatUtils.parseTimeSize(mParameters.mWarmup);
+      mThreadCountResult.setRecordStartMs(recordMs);
+      boolean isRead = FuseIOOperation.isRead(mParameters.mOperation);
+
+      long waitMs = mContext.getStartMs() - CommonUtils.getCurrentMs();
+      if (waitMs < 0) {
+        throw new IllegalStateException(String.format(
+            "Thread missed barrier. Set the start time to a later time. start: %d current: %d",
+            mContext.getStartMs(), CommonUtils.getCurrentMs()));
+      }
+      CommonUtils.sleepMs(waitMs);
+      mStartBarrierPassed = true;
+
+      for (int i = 0; i < mFilesPath.size(); i++) {
+        String filePath = mFilesPath.get(i);
+        while (!Thread.currentThread().isInterrupted() && (!isRead
+            || CommonUtils.getCurrentMs() < mContext.getEndMs())) {
```

Comment:
The logic here is a little complicated ... Fix the next comment and then see how to simplify the logics here

Reply:
if CommonUtils.getCurrentMs() < mContext.getEndMs() for read, then we should break the big for loop

Reply:
> if CommonUtils.getCurrentMs() < mContext.getEndMs() for read, then we should break the big for loop

Thanks for pointing this out!